memdb.go

```diff
@@ -10,11 +10,18 @@ import (
 	"github.com/hashicorp/go-immutable-radix"
 )
 
-// MemDB is an in-memory database.
+// MemDB is an in-memory database providing Atomicity, Consistency, and
+// Isolation from ACID. MemDB doesn't provide Durability since it is an
+// in-memory database.
 //
 // MemDB provides a table abstraction to store objects (rows) with multiple
 // indexes based on inserted values. The database makes use of immutable radix
 // trees to provide transactions and MVCC.
+//
+// Objects inserted into MemDB are not copied. It is **extremely important**
+// that objects are not modified in-place after they are inserted since they
+// are stored directly in MemDB. It is only safe to modify existing values
+// if they have been deleted from MemDB (and the delete transaction committed).
```

Comment:
This is the key bit.

Reply:
I don't think the last statement is true. It's _still_ unsafe to mutate an object even if the delete transaction has comitted since there could be a reader who is still iterating over an older snapshot of the data and reading that object from another goroutine.

I think the only thing that is guaranteed safe usage is "Objects must _never_ be mutated after being passed to `Insert` for the lifetime of the program".

Am I missing some reason this is safe? The only loophole I can see is if it was inserted and deleted in the same transaction and so never was actually visible to another thread.

Reply:
You're right. I'll remove that.